### PR TITLE
Update filtering docs to reflect that particles should not be used with cut regions

### PR DESCRIPTION
**Pull request by jzuhone**
## PR Summary

Cut regions fail for particle-based fields, as pointed out in issue #1976. 

Some work probably needs to be done to decide if we want to support them in cut regions at all, but @chummels made a good suggestion in that issue that we should update the docs to reflect the fact that particle filters should be used instead. This PR does that, and fixes an error on the same page. 

EDIT: We now provide a function that checks if the fields provided to filter a `YTCutRegion` object are mesh fields, and throws an error if not.

### CODE REVIEW

**Review by munkm — APPROVED** [review]
I think these changes are good! 

Maybe we should add a warning log message in the `cut_region` function if it's clear it's being used on a particle field. What do you think?

**Comment by jzuhone** [comment]
@munkm or maybe an error?

**Comment by munkm** [comment]
Either works for me! Since the doc you wrote says it is "unstable" I at least thing users should be warned if it doesn't crash.

**Comment by jzuhone** [comment]
@yt-fido test this please

File: doc/source/analyzing/filtering.rst [hunk 1]
```diff
@@ -118,8 +118,11 @@ The following exclude and include functions are supported:
    - :func:`~yt.data_objects.data_containers.YTSelectionContainer3D.exclude_below` - Exclude values below given value
 
 
-Cut regions can also operate on particle fields, but a single cut region object
-cannot operate on both particle fields and mesh fields at the same time.
+.. warning::
+
+    Cut regions can have unexpected results with particle fields, or often not work
+    at all with some operations. If you want to filter particle fields, see the
+    next section :ref:`filtering-particles` instead.
```
Comment (munkm) [inline doc/source/analyzing/filtering.rst:125]:
```suggestion
    Cut regions are unstable when used onparticle fields, and will currently
    error if you pass a particle field through a cut region. 
    If you want to filter particle fields, see the
    next section :ref:`filtering-particles` instead.
```

**Review by munkm — APPROVED** [review]
Since you're using an error now I'd update the doc too! But this is great and thank you for the changes.

File: doc/source/analyzing/filtering.rst (hunk 1 above)
Comment (jzuhone) [reply doc/source/analyzing/filtering.rst:125]: I've got a slightly modified version of this I'll add

Comment (munkm) [reply doc/source/analyzing/filtering.rst]: Totally cool!

**Review by matthewturk — APPROVED** [review]

**Pull request merged**